strawberry/utils/typing.py

```diff
@@ -249,6 +255,36 @@ def eval_type(
             assert ast_unparse
             type_ = ForwardRef(ast_unparse(parsed))
 
+        # When using forward refs, the whole
+        # Annotated[SomeType, strabwerry.lazy("type.module")] is a forward ref,
+        # and trying to _eval_type on it will fail. Take a different approach
+        # here to resolve lazy types by execing the annotated args and resolving
+        # the type directly.
+        annotated_match = _annotated_re.match(type_.__forward_arg__)
+        if annotated_match:
+            gdict = annotated_match.groupdict()
+            ldict = (localns or {}).copy()
+            # Exec the remaining annotated args to get their real values and
+            # put the result in ldict["args"]
+            exec(f'args = ({gdict["args"]}, )', globalns, ldict)
+            args = ldict["args"]
+
+            for arg in args:
+                if isinstance(arg, StrawberryLazyReference):
```

Comment:
I would prefer to use `not isinstance(...)` to avoid nesting, maybe we could do the same with `if annotated_match` check?

Reply:
I changed it to use `next(...)` to avoid nesting. Should solve this and the issue above